service/collector/trusted_advisor.go

```diff
@@ -1,167 +1,210 @@
 package collector
 
 import (
-	"fmt"
 	"strconv"
-	"sync"
 
 	"github.com/aws/aws-sdk-go/service/support"
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
 	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
 
 	"github.com/giantswarm/aws-operator/client/aws"
 )
 
 const (
-	// serviceLimitCategory is the category returned by Trusted Advisor
-	// for checks related to service limits and usage.
-	serviceLimitCategory = "service_limits"
+	// categoryServiceLimit is the category returned by Trusted Advisor for checks
+	// related to service limits and usage.
+	categoryServiceLimit = "service_limits"
+)
+
+const (
+	indexRegion  = 0
+	indexService = 1
+	indexName    = 2
+	indexLimit   = 3
+	indexUsage   = 4
+)
 
+const (
 	// resourceMetadataLength is the length of resource metadata we expect.
 	resourceMetadataLength = 6
+)
 
-	regionIndex    = 0
-	serviceIndex   = 1
-	limitNameIndex = 2
-	limitIndex     = 3
-	usageIndex     = 4
-
-	accountIDLabel = "account_id"
-	regionLabel    = "region"
-	serviceLabel   = "service"
-	limitNameLabel = "name"
+const (
+	labelAccountID = "account_id"
+	labelRegion    = "region"
+	labelService   = "service"
+	labelName      = "name"
 )
 
 var (
+	getChecksDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_checks_duration",
+		Help:      "Histogram for the duration of Trusted Advisor get checks calls.",
+	})
+	getResourcesDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_resources_duration",
+		Help:      "Histogram for the duration of Trusted Advisor get resource calls.",
+	})
 	serviceLimit *prometheus.Desc = prometheus.NewDesc(
 		prometheus.BuildFQName(Namespace, "", "service_limit"),
 		"Service limits as reported by Trusted Advisor.",
 		[]string{
-			accountIDLabel,
-			regionLabel,
-			serviceLabel,
-			limitNameLabel,
+			labelAccountID,
+			labelRegion,
+			labelService,
+			labelName,
 		},
 		nil,
 	)
 	serviceUsage *prometheus.Desc = prometheus.NewDesc(
 		prometheus.BuildFQName(Namespace, "", "service_usage"),
 		"Service usage as reported by Trusted Advisor.",
 		[]string{
-			accountIDLabel,
-			regionLabel,
-			serviceLabel,
-			limitNameLabel,
+			labelAccountID,
+			labelRegion,
+			labelService,
+			labelName,
 		},
 		nil,
 	)
-
 	trustedAdvisorSupport *prometheus.Desc = prometheus.NewDesc(
 		prometheus.BuildFQName(Namespace, "", "trusted_advisor_supported"),
 		"Gauge describing whether Trusted Advisor is available with your support plan.",
 		nil, nil,
 	)
+)
 
-	trustedAdvisorError = prometheus.NewCounter(prometheus.CounterOpts{
-		Namespace: Namespace,
-		Name:      "trusted_advisor_error_count",
-		Help:      "Counter for the number of errors encountered calling Trusted Advisor.",
-	})
```

Comment:
I removed this one, because I did not see the point. I think we should sort this differently. Also because it is awful with the error handling in the code. I am also not sure which value this feature added so far.

Reply:
Sounds good. We can do this in the collector set going forwards.